src/Payments.sol

```diff
@@ -1239,6 +1242,10 @@ contract Payments is
         // Note: The paymentFee remains in the contract implicitly
         // but is tracked for owner withdrawal
         if (paymentFee > 0) {
+            // Check if this is the first fee for this token
+            if (accumulatedFees[rail.token] == 0) {
```

Comment:
This breaks with withdraw (we can draw the funds down to 0, resetting this).

Reply:
@Stebalien Have fixed this.